src/app/import/components/modal-upload/modal-upload.component.ts

```diff
@@ -0,0 +1,103 @@
+import { AfterViewInit, Component, ElementRef, EventEmitter, Input, OnChanges, OnInit, Output, SimpleChanges, ViewChild } from '@angular/core';
+import { createDefaultModalUploadModal, getSubtitle, getTitle, UploadModals } from './modal-upload-helper';
+import { UploadFileType, UploadOptions } from '../helpers/upload-types';
+import { UploadComponent } from '../upload/upload.component';
+import { UploadHelper } from '../helpers/upload-helper';
+import { UploadStates } from '../../services/s3.enums';
+import { interval } from 'rxjs';
+import { NotificationService } from 'src/app/base/services/notification.service';
+
+@Component({
+  selector: 'kern-modal-upload',
+  templateUrl: './modal-upload.component.html',
+  styleUrls: ['./modal-upload.component.scss']
+})
+export class ModalUploadComponent implements OnInit, AfterViewInit, OnChanges {
+
+  @Input() uploadFileType: UploadFileType;
+  @Input() isModalOpen: boolean;
+  @Input() uploadOptions: UploadOptions;
+  @Input() projectId?: string;
+
+  @Output() closeModalEvent = new EventEmitter();
+  @Output() fileAttached = new EventEmitter<File>();
+  @Output() refetchLookupLists = new EventEmitter<boolean>();
+
+  get UploadFileType(): typeof UploadFileType {
+    return UploadFileType;
+  }
+
+  get UploadStatesType(): typeof UploadStates {
+    return UploadStates;
+  }
+
+  @ViewChild('fileUpload') fileUpload: ElementRef;
+  @ViewChild(UploadComponent) baseComponent: UploadComponent;
+  uploadModals: UploadModals = createDefaultModalUploadModal();
+  file: File | null = null;
+  uploadHelper: UploadHelper;
+  title: string;
+  subTitle: string;
+
+  constructor() { }
+
+  ngOnInit(): void {
+    NotificationService.subscribeToNotification(this, {
+      projectId: this.projectId,
+      whitelist: ['file_upload'],
+      func: this.handleWebsocketNotification
+    });
+  }
+
+  ngAfterViewInit(): void {
+    this.uploadHelper = new UploadHelper(this.baseComponent, this.baseComponent.recordNewUploadHelper, this.baseComponent.recordAddUploadHelper, this.baseComponent.existingProjectUploadHelper, this.baseComponent.lookupListsUploadHelper);
+  }
+
+  ngOnChanges(changes: SimpleChanges): void {
+    if (changes.isModalOpen) {
+      this.uploadModals.uploadFile.open = changes.isModalOpen.currentValue;
+      this.title = getTitle(this.uploadFileType);
+      this.subTitle = getSubtitle(this.uploadFileType);
```

Comment:
why are these reset if the open state changes?

Reply:
Moved them to on init